packages/material-ui-lab/src/Slider/Slider.js

```diff
@@ -418,14 +430,30 @@ class Slider extends React.Component {
       [classes.vertical]: vertical,
     });
 
-    const thumbClasses = classNames(classes.thumb, commonClasses);
-
     const trackProperty = vertical ? 'height' : 'width';
     const thumbProperty = vertical ? 'top' : 'left';
     const inlineTrackBeforeStyles = { [trackProperty]: this.calculateTrackBeforeStyles(percent) };
     const inlineTrackAfterStyles = { [trackProperty]: this.calculateTrackAfterStyles(percent) };
     const inlineThumbStyles = { [thumbProperty]: `${percent}%` };
 
+    /** Start Thumbnail Icon Logic Here */
```

Comment:
Could you change references to Thumbnail to something less confusing? a thumbnail and a (Material) thumb are two different things.

Reply:
`thumbIcon`?

Reply:
thumbIcon sounds nice. Should this be the external prop too. e.g, right now it is only `thumb`.
```
<Slider {...otherProps} thumb={} />
```
Should I change this to `thumbIcon` too, along with inside the code as well.

And in Slider.js file
use `thumbIcon` as `ThumbIcon` because I need to clone the element and pass some props to it.

Reply:
Done, I have made changes. Can you please review. Again thank you for taking your busy time out to review my code. This really means a lot.